### PR TITLE
[WIP] V1.2: Add documentation and do some minor refactoring

### PR DESCRIPTION
**Pull request by ZHTang29**
Add documentation for new classes and their relevant methods in model package; Refactor part of ModelManager methods

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -5,16 +5,30 @@ title: Developer Guide
 * Table of Contents
 {:toc}
 
+## **Appendix: Requirements**
+
+###Product scope
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:10]: Add a space after ### so it applies
`### Product scope`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -5,16 +5,30 @@ title: Developer Guide
...
+**Target user profile:**
+* has a need to manage a large number of recipes and their relevant information
+* prefers desktop applications to other types
+* ability to type fast
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:14]: types fast

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -5,16 +5,30 @@ title: Developer Guide
 * Table of Contents
 {:toc}
 
+## **Appendix: Requirements**
+
+###Product scope
+**Target user profile:**
+* has a need to manage a large number of recipes and their relevant information
+* prefers desktop applications to other types
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:13]: prefers desktop applications to mobile

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -121,6 +135,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (poss
   
     Use case resumes from step 2.
 
+#### **Use case: Ask for help**
+
+**MSS**
+
+1. Chef requests for help
+2. Book shows a list of all commands, what they do, and example usage
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:143]: shows a list of the descriptions and example usages of all commands

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -5,16 +5,30 @@ title: Developer Guide
...
+* has a need to manage a large number of recipes and their relevant information
+* prefers desktop applications to other types
+* ability to type fast
+* prefers typing to mouse interactions
+
+**Value proposition:** manage recipes more conveniently and quickly as opposed to a typical mouse app
+
 ### User stories
 
 Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (possible but unlikely to add) - `*`
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:21]: `Low(unlikely to have)` to standardise to the docs

File: docs/UserGuide.md [hunk 6]
```diff
@@ -43,174 +44,76 @@ features in mind so you can always find the recipe you need! Make use of ***RIZZ
...
-> :bulb: Tip: It is okay to not add a list of ingredients when you first upload
-> your recipe (you can add them later!).
-> We do however recommend you add quantity when you add ingredients
-> The same applies for steps.
+> :bulb: Tip: A recipe can have any number of steps, tags and ingredients (including 0)!
+> :bulb: Tip: It is only compulsory to include the recipe name when you first add the recipe into recipe book!
+> Although optional, we do however still suggest adding in the other fields if possible for your own convenience.
+> We recommend you add in your own units of measurement for the optional fields! (i.e g/kg for ingredient weights)
+> If you have multiple inputs for tags, steps or ingredients, treat each input as a separate field (i.e s/Step1 s/step2)
+> Each field does not have to be inputted in order (/p can come before /d etc.)
```
Comment (JamesLiuZX) [inline docs/UserGuide.md:55]: input is a more widely-used term for past tense

File: docs/UserGuide.md [hunk 7]
```diff
@@ -43,174 +44,76 @@ features in mind so you can always find the recipe you need! Make use of ***RIZZ
...
-```
-
-Lists all recipes that are in the storage, in the chronological order they were
-added. Depending on the size of the window, may add multiple columns to display
+Format:
+`list`
+
+> Lists all recipes that are in the storage, in the chronological order they were
+added. 
+> Depending on the size of the window, may add multiple columns to display
```
Comment (JamesLiuZX) [inline docs/UserGuide.md:75]: "one may add"

**Review by JamesLiuZX — COMMENTED** [review]
Overall great PR! The enthusiasm and personality in the UG can be shown through the relatable rhetorics in feature descriptions and good use of exclamation marks, as well as various lightbulb and note emojis. A bit of fixing on the grammar side and it's good to go. LGTM 👍

**Review by JamesLiuZX — APPROVED** [review]

File: docs/UserGuide.md [hunk 8]
```diff
@@ -43,174 +44,76 @@ features in mind so you can always find the recipe you need! Make use of ***RIZZ
...
-- The index **must be a positive integer** 1, 2, 3, …
+Unsure or unable to remember our list of commands and how to format your inputs? Fret not, as simply inputting
+`help` will save you from your woes.
 
-Example of usage:
+Format:
+`help`
 
-`delete 1`
+> :bulb: Tip: `help` a link to this user guide, which is a quick way to refresh your memory on any command that might have slipped your mind!
```
Comment (filbertphang) [inline docs/UserGuide.md:108]: typo (?): "`help` a link to this UG..." -> "`help` links to this UG..."

File: docs/UserGuide.md [hunk 9]
```diff
@@ -241,13 +144,14 @@ _Details coming soon ..._
...
-| **Exit**   | `exit`                                     |   
+| Action     | Format, Examples                                                                                                                                                                                                      |
+|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Add**    | `add n/RECIPE_NAME d/RECIPE_DURATION p/RECIPE_PORTION [t/RECIPE_TAGS] [s/RECIPE_STEPS] [i/RECIPE_INGREDIENTS]`<br/><br/> e.g., `add n/Chicken Noodles d/20 minutes p/1-2 people i/chicken thigh i/noodle i/soy sauce` |                                                                                                                                                             |            |                                      |
```
Comment (filbertphang) [inline docs/UserGuide.md:149]: need to update the command format here also for `add` and `delete` (wrt the changes specified above)

**Review by filbertphang — COMMENTED** [review]
minor changes needed with the user guide, but overall excellent work with the documentation 👍  can be merged after those issues are fixed

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -121,6 +135,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (poss
   
     Use case resumes from step 2.
 
+#### **Use case: Ask for help**
+
+**MSS**
+
+1. Chef requests for help
+2. Book shows a list of the descriptions and example usages of all commands
```
[the remark this answers is not in the document]
Comment (ZHTang29) [reply docs/DeveloperGuide.md:143]: Yeah my bad, changed it for the User Guide but not for the Developer Guide. Will update this behavior for this iteration.

File: docs/UserGuide.md [hunk 11]
```diff
@@ -43,174 +44,76 @@ features in mind so you can always find the recipe you need! Make use of ***RIZZ
 Come up with a new innovative recipe and want to store it for future reference,
 and want to classify it by ingredients? Simply run the `add` command, and follow the prompts!
 
-Format: 
-```text
-add RECIPE_NAME
-```
+Format:
+`add n/RECIPE_NAME d/RECIPE_DURATION p/RECIPE_PORTION [t/RECIPE_TAGS] [s/RECIPE_STEPS] [i/RECIPE_INGREDIENTS]`
```
[the remark this answers is not in the document]
Comment (ZHTang29) [reply docs/UserGuide.md:48]: Good spot, edited on my end.

**Pull request merged**